lib/Error.hs

```diff
@@ -11,5 +11,6 @@ import qualified Data.Text as T
 data CofferError =
   MarshallingFailed
   | ConnectError
+  | BackendNotFound T.Text
```

Comment:
These 3 definitions should be changed from `Text` to `BackendName`:

* `CofferError.BackendNotFound`
* `Config.mainBackend`
* `Backend._name`

Reply:
Done